src/sources/file.rs

```diff
@@ -511,6 +511,11 @@ pub fn file_source(
         return Box::pin(future::ready(Err(())));
     }
 
+    let exclude_patterns = config
+        .exclude
+        .iter()
+        .map(|path_buf| path_buf.iter().collect::<std::path::PathBuf>())
+        .collect::<Vec<PathBuf>>();
```

Comment:
I don't think this is right. It'll attempt to interpret the exclude patterns as paths, but they are glob patterns. Am I missing something?

Reply:
You may check this method [components](https://doc.rust-lang.org/src/std/path.rs.html#2645-2689).
The path iter iterates over its components. The components method just does a simple normalization and won't affect glob patterns.

Reply:
Aha, apologies, I missed that this was just parsing into path components. 👍